src/views/connectionUI.ts

```diff
@@ -85,32 +85,32 @@ export class ConnectionUI {
 	 * @returns The connectionInfo choosen or created from the user, or undefined if the user cancels the prompt.
 	 */
 	public async promptForConnection(ignoreFocusOut: boolean = false): Promise<IConnectionInfo | undefined> {
```

Comment:
This method won't work correctly anymore - it always returns undefined. Situations like this you need to keep the Promise and resolve/reject because those are done in callbacks. Removing the Promise means it'll go set up the callbacks but then never actually wait for them to be called and instead return immediately.

Reply:
Keeping it the same is leading to test failures, as promise is not resolving with current implementation. Any thoughts?

Reply:
Oof, yeah so the root cause of this is your change to make loadAllConnections async (because you persist the profiles when you update the connection encrypt field). 

promptForConnection calls getPickListItems which calls loadAllConnections. Previously this was all sync calls so it would complete all the calls in promptForConnection before returning the promise. But now with the async call it hits the await and then moves on to other tasks, which includes returning the created promise (which isn't done setting up all the calls yet such as the onDidChangeSelection callback). 

The test then moves on to firing the event, but because it hasn't hooked up the onDidChangeSelection callback yet that code is never hit so it doesn't resolve the promise like expected and thus the test times out. 

Here's the order of events:
1. Test calls promptForConnection
2. It creates the Promise which then in the constructor calls the callback
3. In the callback we call getPickListItems - because it's a promise it puts it in the task queue and moves on to other stuff
4. promptForConnection gets control again and returns the promise (which is still in the process of being set up)
5. The test fires the event
...some time later the call to getPickListItems completes and then it finally finishes setting up the promise (which includes hooking up the onDidChangeSelection callback)

With the way this code is currently written we pretty much can't have any awaitable code inside promptForConnection because of this.

My suggestion to fix this is don't check/update the encrypt value every single time we call loadAllConnections. This seems like it should just be a one-time thing anyways (done on extension load) so maybe you can make a separate call to just fix up all the encrypt values on extension load instead of doing it every single time that we call loadAllConnections?